artiq/coredevice/phaser.py

```diff
@@ -250,6 +250,11 @@ def __init__(self, dmgr, channel_base, miso_delay=1, tune_fifo_offset=True,
         self.channel = [PhaserChannel(self, ch, trf)
                         for ch, trf in enumerate([trf0, trf1])]
 
+    @staticmethod
+    def get_rtio_channels(channel_base, **kwargs):
+        return [(channel_base, None), (channel_base + 1, None),
```

Comment:
* You can write that shorter
* This is wrong for `self.gw_rev != PHASER_GW_BASE`

Reply:
What is the difference in case `self.gw_rev != PHASER_GW_BASE` ?

Btw, Implementation of this difference may require dynamic channel map, since the `self.gw_rev` couldn't be constructed without accessing hardware.

Reply:
According to this snippet, there will be no issue if `Phaser` will ultimately take 5 channels for name resolving.

Reply:
There is no problem with autoprobing in core device drivers. It's just the fragile and incoherent design here that is unable to cope with it.

Reply:
That's terse. Care to elaborate? The (local) RTIO map is completely defined at gateware build time.
Using that instead of the various hacks to enumerate again later would dramatically simplify things across the landscape (from AFWS all the way to this code here).